### PR TITLE
Adds grabbing yourself, fixes/changes some more wound things

### PR DESCRIPTION
**Pull request by Ryll-Ryll**
## About The Pull Request
There's not really a lot of things you can do to deal with massive bleeding on singular limbs if you don't have any medical/improvised supplies that fit the bill, so I thought it'd be cool and thematic if you could grasp your gushing limbs to try keeping yourself together. 

[](https://i.imgur.com/ThKdu3G.png)

By clicking yourself with grab intent while targeting a limb of yours that's bleeding, you'll "grab" onto it with your free hand to apply pressure. This takes 1.5 seconds to do, takes up a hand while you do so, and applies a slowdown while you're holding it. In return, that limb will bleed 30% less while you're holding it (though it won't clot any faster). This stacks with the 25% reduction in bleeding you get from being horizontal, so laying down clutching your gushing gut while waiting for help is now both thematic _and_ viable!

### Other fixes and changes
I also made some other changes to wounds of various importance. Here's a quick list

- Most wound treatments now respect thick clothing/materials, so you can no longer suture slashes through spacesuits.
- Using sutures on slashes and pierces now actually use up sutures
- Coagulant has been nerfed again to reduce bleeding less for every wound it's reducing bleeding on. This'll still need more love and attention later, but I need to find a good middle point where medipens are good for bleeding in an emergency without being a way to sidestep any and all bleeding wounds in one go. 
- Infectious zombies (romerol) can now be wounded as intended
- I added a recipe for improvised coagulant in the Wounds 2 PR using baked pulped banana peels and salglu, but then realized that recipe sucked and was overly complicated. You can now drink straight up ground/juiced banana peels to help close up slashes and piercings. Yu~~ck~~m!
- You can no longer repeatedly disembowel people who have already lost their internal organs

The self-grasp code is admittedly pretty sketchy, but I'm sure it can be cleaned up as needed.

## Why It's Good For The Game
Adds a very basic improvised method for staving off death by bloodloss while waiting for treatment, make wound treatments more consistent with how healing items usually work, fixes odd/buggy behavior

## Changelog
:cl: Ryll/Shaps
add: You can now grab your own bleeding limbs to slow your bloodloss! Target your bleeding limb and click on yourself with grab intent and an open hand. This won't speed the clotting, but you'll lose blood more slowly.
add: You can now juice/pulp banana peels to act as makeshift coagulant. hahaha ewwwwww
fix: Most wound treatments (besides manipulating dislocated limbs) now respect thick clothing, so you can no longer suture through space suits.
fix: Romerol zombies can now be wounded properly
fix: Using sutures to treat slashes and pierces now correctly consumes the sutures
fix: You can no longer repeatedly disembowel people who have already lost their guts.
/:cl:

### CODE REVIEW

**Comment by Krysonism** [comment]
Why should you be able to deal with massive bleeding if you don't have any of the following: epi pen, gauze, tape, sutures, improvised coagulant, cautery, welder, lighter, any other hot item, catgirl, etc?

When are people supposed to bleed out?

**Comment by Ryll-Ryll** [comment]
Mind that epipens took a sizeable nerf in this PR with the coagulant nerf, and I also sliightly lowered the clot rate for severe slashes. Holding your limb has a sizeable slowdown and takes up a hand slot, only affects one (or two I guess) limbs, and only reduces how much blood you lose per tick rather than actually close the cuts faster. I'm hoping that by adding a very basic method to reduce bloodloss, that I can further reduce the effectiveness of things like epipens

**Comment by Krysonism** [comment]
> Mind that epipens took a sizeable nerf in this PR with the coagulant nerf, and I also sliightly lowered the clot rate for severe slashes. Holding your limb has a sizeable slowdown and takes up a hand slot, only affects one (or two I guess) limbs, and only reduces how much blood you lose per tick rather than actually close the cuts faster. I'm hoping that by adding a very basic method to reduce bloodloss, that I can further reduce the effectiveness of things like epipens

Sounds great

**Comment by optimumtact** [comment]
Do we not have tourniquet's ?

You'd think everyone's emergency kit would come with one

**Comment by Ryll-Ryll** [comment]
I'd only consider tourniquets if I added progressive dismemberment where you can lose parts of your limb to render them partly useless, so there'd be a balance between how much of the limb could be salvaged/still used vs how much blood you'd lose with some gauze vs a tourniquet. Otherwise, medkits are already pretty stuffed

**Comment by Couls** [comment]
We already have disabled limbs, perhaps you can expand the system behind mangling?

**Comment by necromanceranne** [comment]
How does this interact with autoaggrograbs like martial arts? Are CQC users able to grasp their wounds? Or (although I imagine wounding them is unlikely/impossible) leather golems?

**Comment by ATH1909** [comment]
> Most wound treatments now respect thick clothing/materials, so you can no longer suture slashes through spacesuits.

Make sure that this won't make it impossible for pierceimmune races that can still receive wounds (such as, say, skeletons) to heal themselves.

> Coagulant has been nerfed again to reduce bleeding less for every wound it's reducing bleeding on. This'll still need more love and attention later, but I need to find a good middle point where medipens are good for bleeding in an emergency without being a way to sidestep any and all bleeding wounds in one go.

Maybe, instead of having coagulant directly heal your bleeding wounds, it could just halve the blood you lose from said bleeding wounds while it's in your system?

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm [hunk 1]
```diff
@@ -1387,8 +1388,13 @@
 			var/obj/item/organ/heart/our_heart = M.getorganslot(ORGAN_SLOT_HEART)
 			our_heart.applyOrganDamage(1)
 
-// can be synthesized on station rather than bought. made by grinding a banana peel, heating it up, then mixing the banana peel powder with salglu
-/datum/reagent/medicine/coagulant/weak
-	name = "Synthi-Sanguirite"
-	description = "A synthetic coagulant used to help bleeding wounds clot faster. Not quite as effective as name brand Sanguirite, especially on patients with lots of cuts."
-	clot_coeff_per_wound = 0.8
+// i googled "natural coagulant" and a couple of results came up for banana peels, so after precisely 30 more seconds of research, i now dub grinding banana peels good for your blood
+/datum/reagent/medicine/coagulant/banana_peel
```
Comment (ATH1909) [inline code/modules/reagents/chemistry/reagents/medicine_reagents.dm:1392]: why not just use the existing banana juice reagent for this? other juices already heal things like brute, burn, tox, and oxy damage, so the schtick of banana juice could be healing cut wounds (and also directly healing clowns, like it already does).

Comment (Ryll-Ryll) [reply code/modules/reagents/chemistry/reagents/medicine_reagents.dm:1392]: Read the comment ath, it's right there. Just read it. It's there for a reason.

Comment (tmtmtl30) [reply code/modules/reagents/chemistry/reagents/medicine_reagents.dm:1392]: that doesn't mean it isn't bloat-y

**Comment by Ryll-Ryll** [comment]
I don't foresee that being an issue

**Comment by optimumtact** [comment]
@tmtmtl30 it seems kind of odd to complain about the increase of complexity when that was clearly signalled as the reason to introduce them in the first place?

The whole point was to add things you can't just fix with your standard set of core chems, making people with knowledge able to appear more effective and skillful.

**Comment by GuyonBroadway** [comment]
Now make it so I can grab my crotch, pivot 360 and moonwalk down the corridor.

File: code/modules/mob/living/carbon/human/examine.dm [hunk 2]
```diff
@@ -283,6 +286,12 @@
 				bleed_text += " incredibly quickly"
 
 			bleed_text += "!</B>\n"
+
+		for(var/i in bodyparts)
```
Comment (Rohesie) [inline code/modules/mob/living/carbon/human/examine.dm:290]:
```suggestion
		for(var/i in grasped_limbs)
```
This is what you intended here, right?
If so, checking `grasped_by` again is not necessary, right?

File: code/modules/mob/living/carbon/human/examine.dm [hunk 3]
```diff
@@ -253,11 +253,14 @@
...
 	else if(is_bleeding())
 		var/list/obj/item/bodypart/bleeding_limbs = list()
+		var/list/obj/item/bodypart/grasped_limbs = list()
 
 		for(var/i in bodyparts)
 			var/obj/item/bodypart/BP = i
 			if(BP.get_bleed_rate())
 				bleeding_limbs += BP
+			if(BP.grasped_by)
+				grasped_limbs += BP
```
Comment (Rohesie) [inline code/modules/mob/living/carbon/human/examine.dm:263]:
```suggestion
		var/list/obj/item/bodypart/grasped_limbs

		for(var/i in bodyparts)
			var/obj/item/bodypart/BP = i
			if(BP.get_bleed_rate())
				bleeding_limbs += BP
			if(BP.grasped_by)
				LAZYADD(grasped_limbs, BP)
```

File: code/modules/mob/living/carbon/carbon_defense.dm [hunk 4]
```diff
@@ -498,3 +498,64 @@
...
+
+/// The limb or the whole damn person we were grasping got deleted, so we don't care anymore
+/obj/item/grip_self/proc/qdel_void()
+	qdel(src)
+
+/// We're trying to grasp, but we can only do so if we have a bodypart on the zone we're targeting, and said bodypart is bleeding
+/obj/item/grip_self/proc/try_grasp(mob/living/carbon/attempted_grasper)
+	if(!istype(attempted_grasper))
+		qdel()
+		return
```
Comment (Rohesie) [inline code/modules/mob/living/carbon/carbon_defense.dm:542]:
```suggestion
	if(!istype(attempted_grasper))
		stack_trace("[src] attempted to try_grasp() with [istype(attempted_grasper, /datum) ? attempted_grasper.type : isnull(attempted_grasper) ? "null" : attempted_grasper] attempted_grasper")
		qdel()
		return
```

File: code/modules/mob/living/carbon/carbon_defense.dm [hunk 5]
```diff
@@ -498,3 +498,64 @@
...
+
+	grasped_part = attempted_grasper.get_bodypart(attempted_grasper.zone_selected)
+	if(!grasped_part?.get_bleed_rate() || !attempted_grasper.put_in_active_hand(src))
+		qdel()
```
Comment (Rohesie) [inline code/modules/mob/living/carbon/carbon_defense.dm:546]:
```suggestion
		qdel(src)
```
Might want to put the `get_bleed_rate()` check before creating the object.

File: code/modules/mob/living/carbon/carbon_defense.dm [hunk 6]
```diff
@@ -498,3 +498,64 @@
...
+		return
+
+	to_chat(attempted_grasper, "<span class='warning'>You try grasping at your [grasped_part.name], trying to stop the bleeding...</span>")
+	if(!do_after(attempted_grasper, 1.5 SECONDS))
+		qdel()
```
Comment (Rohesie) [inline code/modules/mob/living/carbon/carbon_defense.dm:551]:
```suggestion
		qdel(src)
```

File: code/modules/mob/living/carbon/carbon_defense.dm [hunk 7]
```diff
@@ -498,3 +498,64 @@
...
+		return
+
+	user = attempted_grasper // if we have a user, we know we were successful
+	grasped_part.grasped_by = src
+	RegisterSignal(user, COMSIG_PARENT_QDELETING, .proc/qdel_void)
+	RegisterSignal(grasped_part, COMSIG_PARENT_QDELETING, .proc/qdel_void)
+
+	user.visible_message("<span class='danger'>[user] grasps at [user.p_their()] [grasped_part.name], trying to stop the bleeding.</span>", "<span class='notice'>You grab hold of your [grasped_part.name] tightly.</span>", vision_distance=COMBAT_MESSAGE_RANGE)
+	playsound(src.loc, 'sound/weapons/thudswoosh.ogg', 50, TRUE, -1)
```
Comment (Rohesie) [inline code/modules/mob/living/carbon/carbon_defense.dm:560]:
```suggestion
	playsound(loc, 'sound/weapons/thudswoosh.ogg', 50, TRUE, -1)
```

File: html/changelog.html [hunk 8]
```diff
@@ -51,6 +51,11 @@
 		
 		-->
 		<div class="commit sansserif">
+			<h2 class="date">03 August 2020</h2>
```
Comment (optimumtact) [inline html/changelog.html:54]: can you not

**Comment by Ryll-Ryll** [comment]
Ackkkk

**Comment by Ryll-Ryll** [comment]
ugh

**Comment by Ryll-Ryll** [comment]
I'll deal with this mess tomorrow

File: code/__HELPERS/hearted.dm [hunk 9]
```diff
@@ -6,10 +6,10 @@
 
 	for(var/i in GLOB.joined_player_list)
 		var/mob/check_mob = get_mob_by_ckey(i)
-		if(!check_mob || !check_mob.mind || !check_mob.client)
+		if(!check_mob?.mind || !check_mob.client)
 			continue
 		// maybe some other filters like bans or whatever
-		check_mob.query_heart(1)
+		INVOKE_ASYNC(check_mob, /mob.proc/query_heart, 1)
```
Comment (Rohesie) [inline code/__HELPERS/hearted.dm:12]: This is from the other PR, right?

File: html/changelogs/.all_changelog.yml [hunk 10]
```diff
@@ -41373,209 +41373,22 @@ DO NOT EDIT THIS FILE BY HAND!  AUTOMATICALLY GENERATED BY ss13_genchangelog.py.
   zxaber:
   - tweak: The NtOS home screen has been updated with disk and ID eject buttons.
 2020-08-02:
-  ArcaneMusic:
```
Comment (Rohesie) [inline html/changelogs/.all_changelog.yml:41376]: Disable the actions in your repository to avoid this kind of thing.
And revert the changelog compile commits to easily undo these changes.

File: code/modules/shuttle/emergency.dm [hunk 11]
```diff
@@ -384,7 +384,7 @@
 				launch_status = ENDGAME_LAUNCHED
 				setTimer(SSshuttle.emergencyEscapeTime * engine_coeff)
 				priority_announce("The Emergency Shuttle has left the station. Estimate [timeLeft(600)] minutes until the shuttle docks at Central Command.", null, null, "Priority")
-				SSticker.poll_hearts()
+				INVOKE_ASYNC(SSticker, /datum/controller/subsystem/ticker.proc/poll_hearts)
```
Comment (Rohesie) [inline code/modules/shuttle/emergency.dm:387]: Other PR bleeding into this one

File: code/modules/surgery/bodyparts/_bodyparts.dm [hunk 12]
```diff
@@ -87,6 +87,8 @@
 	var/generic_bleedstacks
 	/// If we have a gauze wrapping currently applied (not including splints)
 	var/obj/item/stack/current_gauze
+	/// If something is currently grasping this bodypart and trying to staunch bleeding (see [/obj/item/grasp_self])
+	var/obj/item/self_grasp/grasped_by
```
Comment (Rohesie) [inline code/modules/surgery/bodyparts/_bodyparts.dm:91]: Is this being nulled on `Destroy()` ?

File: html/changelog.html [hunk 13]
```diff
@@ -51,6 +51,11 @@
...
+			<h3 class="author">ArcaneMusic updated:</h3>
+			<ul class="changes bgimages16">
+				<li class="rscadd">When 2 plants are crossbreeding, if they share a reagent, one plant can now accept and upgrade their production % from the other.</li>
+				<li class="bugfix">Cabbage no longer wild-mutates into producing humans as a byproduct</li>
```
Comment (Rohesie) [inline html/changelog.html:58]:
```suggestion
```

File: code/modules/mob/living/carbon/carbon_defense.dm [hunk 14]
```diff
@@ -498,3 +498,77 @@
...
+
+	grasped_part = attempted_grasper.get_bodypart(attempted_grasper.zone_selected)
+	if(!grasped_part?.get_bleed_rate())
+		qdel(src)
+		return
+
+	if(attempted_grasper.active_hand_index == grasped_part.held_index)
+		to_chat(attempted_grasper, "<span class='danger'>You can't grasp your [grasped_part.name] with itself!</span>")
+		qdel(src)
+		return
```
Comment (Rohesie) [inline code/modules/mob/living/carbon/carbon_defense.dm:555]: These two checks seem common enough to be better done before creating the item. Creating and deleting a million items does put a certain strain on the GC

File: html/changelog.html (hunk 13 above)
Comment (Ryll-Ryll) [reply html/changelog.html:58]: oh my god this again

**Comment by Ryll-Ryll** [comment]
i hate the changelog i'm redoing this in another PR
